### PR TITLE
chore(CI): Set up next branch workflow WEB-970

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview
- Publishes next branch whenever merged in, no PR required.
- Deploy is handled by netlify.

### PR Checklist

- [ ] Related to Abstract designs:
- [x] Related to JIRA ticket: WEB-970
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1b273ebc1a546568f22728--codecademy-styleguide.netlify.app>https://5f1b273ebc1a546568f22728--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f1b273ebc1a546568f22728>Deploy Logs</a></p>

File: .circleci/config.yml [hunk 1]
```diff
@@ -151,6 +151,20 @@ jobs:
...
+    steps:
+      - *skip_on_automated_commit
+      - *add_github_ssh_key
+      - *set_git_user
+      - *restore_repo
+      - checkout
+      - *set_npm_token
+      - run:
+          name: Lerna Publish
+          command: yarn lerna publish --yes --conventional-commits --changelog-preset conventionalcommits --include-merged-tags --create-release=github --npm-tag next
```
Comment (JoshuaKGoldberg) [inline .circleci/config.yml:166]: This looks like it should work like we want it, I think - I'm curious how that versioning story is going to go? Seems reasonable to try to me!

We should be careful in the next few monolith Gamut bumps to make sure the `next`-tagged versions can't be easily accidentally installed...

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1f29b931526db57aa5599d--gamut-preview.netlify.app>https://5f1f29b931526db57aa5599d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f1f29b931526db57aa5599d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1f2b6426741cbb00b04a9a--gamut-preview.netlify.app>https://5f1f2b6426741cbb00b04a9a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f1f2b6426741cbb00b04a9a>Deploy Logs</a></p>

**Pull request merged**